code/modules/mob/living/carbon/life.dm

```diff
@@ -11,4 +11,65 @@
 		else
 			if (internals)
 				internals.icon_state = "internal0"
-	return
\ No newline at end of file
+	return
+
+/mob/living/carbon/proc/handle_nausea()
+	if(!stat)
+		if(getToxLoss() >= 40)
+			nausea++
+		if(nausea > 0) //so prob isn't rolling on every tick
+			if(prob(6)) //slowly reduce nausea over time
+				nausea--
+			if(nausea >= 12) //not feeling so good
+				if(prob(8))
+					visible_message("<font color='green'>[src] retches!</font>", \
+							"<font color='green'><b>you retch!</b></font>")
+			if(nausea >= 25) //vomiting
+				Stun(5)

```

Comment:
TBH it should probably do a shorter stun if you have enough nutrition (or a longer stun if you have too little nutrition)